dbms/src/DataTypes/DataTypeEnum.cpp

```diff
@@ -397,11 +402,38 @@ static DataTypePtr create(const ASTPtr & arguments)
     return std::make_shared<DataTypeEnum>(values);
 }
 
+static DataTypePtr create(const ASTPtr & arguments)
+{
+    if (!arguments || arguments->children.empty())
+        throw Exception("Enum data type cannot be empty", ErrorCodes::EMPTY_DATA_PASSED);
+
+    /// Children must be functions 'equals' with string literal as left argument and numeric literal as right argument.
+    for (const ASTPtr & child : arguments->children)
+    {
+        checkASTStructure(child);
+
+        const auto * func = child->as<ASTFunction>();
+        const auto * value_literal = func->arguments->children[1]->as<ASTLiteral>();
+
+        if (!value_literal)
+            throw Exception(
+                    "Elements of Enum data type must be of form: 'name' = number, where name is string literal and number is an integer",
+                    ErrorCodes::UNEXPECTED_AST_STRUCTURE);
+
+        Int64 value = value_literal->value.get<Int64>();
```

Comment:
It's better to check that `value_literal->value.getType() == Field::Types::UInt64 || value_literal->value.getType() == Field::Types::Int64`  as in function above. Because `get` doesn't check type of Field and will return garbage for other types.